x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTask.java

```diff
@@ -50,13 +50,15 @@
     public static final ParseField MAX_WRITE_BUFFER_SIZE = new ParseField("max_write_buffer_size");
     public static final ParseField MAX_RETRY_DELAY = new ParseField("max_retry_delay");
     public static final ParseField POLL_TIMEOUT = new ParseField("poll_timeout");
-    public static final ParseField RECORDED_HISTORY_UUID = new ParseField("recorded_history_uuid");
+    public static final ParseField RECORDED_LEADER_HISTORY_UUID = new ParseField("recorded_leader_history_uuid");
```

Comment:
Why are these properties of the Task - they are properties of the index (meta data)? we should get them from there?

Reply:
Good point. We can remove them here and then make the IndexMetaData of the follower available in `ShardFollowNodeTask`. Are you ok if I do this in a follow up? (it affect the UUIDs of both leader and follower index)